contracts/src/v0.1/Prepayment.sol

```diff
@@ -254,25 +281,113 @@ contract Prepayment is Ownable, IPrepayment, ITypeAndVersion {
     /**
      * @inheritdoc IPrepayment
      */
-    function createAccount() external returns (uint64) {
+    function getAccountDetail(
+        uint64 accId
+    ) external view returns (uint256, uint256, uint256, uint256) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getAccountDetail();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function getSubscriptionPaid(uint64 accId, uint256 index) external view returns (bool) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getSubscriptionPaid(index);
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function isValidReq(uint64 accId) external view returns (bool) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.isValidReq();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function getFeeRatio(uint64 accId) external view returns (uint256) {
+        Account account = sAccIdToAccount[accId];
+        if (address(account) == address(0)) revert InvalidAccount();
+        return account.getFeeRatio();
+    }
+
+    /**
+     * @inheritdoc IPrepayment
+     */
+    function createAccount(IAccount.AccountType accType) external returns (uint64) {
```

Comment:
I think we cannot let anybody to choose whatever account they want to have. There has to be some rules, please think about that.

Reply:
How about we separate `createAccount()` functions for the 3 types: `FIAT_SUBSCRIPTION`, `KLAY_SUBSCRIPTION`, `KLAY_DISCOUNT`